hack/generate-jobs.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generate job configuration files."""
```

Comment:
I have mixed feelings about putting this under `test-infra/hack/generate-jobs.py`. Do we expect that other people, potentially from other SIGs than SIG Node, are going to use it?

If yes, then we need more documentation on how to use it and SIG Testing has to agree to maintain this for everyone.

If no, then it might be better to keep it local to the sig-node directory and only change the top-level rules so that they pick up SIG-owned generate and verify scripts.
 
 cc @BenTheElder @aojea @dims 
 
Personally I think this is the right direction for Kubernetes, so perhaps it is in the right place.

Reply:
I'm ok with either way. Anyway we're going to start with DRA and hopefully with SIG-node.

Reply:
I am ok to leave it where it is. if it gets contentious then we can move it.